frontend/src/dashboard/CreateClassForm.js

```diff
@@ -0,0 +1,75 @@
+import React, {useState} from 'react';
+import {Button, Form, Message} from 'semantic-ui-react';
+
+import {classCreateRequestor} from '../requests/requestBuilder';
+
+export const CreateClassForm = props => {
+  const [isFormOpen, toggleForm] = useState(false);
+  const [className, setClassName] = useState('');
+  const [createSuccess, showMessage] = useState(false);
+  const [createFail, showError] = useState(false);
+
+  const createClass = () => {
+    for(var c in props.classes){
+      if(props.classes[c].classname == className){
+        showError(true);
+        return;
+      }
+    }
+    classCreateRequestor(className, localStorage.getItem('userid'))
+      .then(response => {
+        props.refreshClasses();
+        showMessage(true);
+        showError(false);
+      })
+      .catch(error => {
+        console.error(error);
+      });
+    toggleForm(false);
```

Comment:
What is toggle form supposed to do?

Reply:
open or close the form

Reply:
so here, u r closing the form while create class requester is running. and not after it finishes.
do you want the form to be closed regardless or only on success? if regardless then it's fine now. if u want it on success then u should put it in the then block